content/en/docs/ops/diagnostic-tools/istioctl-analyze/snips.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+# shellcheck disable=SC2034,SC2153,SC2155,SC2164
+
+# Copyright Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+####################################################################################################
+# WARNING: THIS IS AN AUTO-GENERATED FILE, DO NOT EDIT. PLEASE MODIFY THE ORIGINAL MARKDOWN FILE:
+#          docs/ops/diagnostic-tools/istioctl-analyze/index.md
+####################################################################################################
+
+snip_analyze_all_namespaces() {
+istioctl analyze --all-namespaces
+}
+
+! read -r -d '' snip_analyze_all_namespaces_sample_response <<\ENDSNIP
+Warn [IST0102](Namespace default) The namespace is not enabled for Istio injection. Run 'kubectl label namespace default istio-injection=enabled' to enable it, or 'kubectl label namespace default istio-injection=disabled' to explicitly mark it as not needing injection
+ENDSNIP
+
+snip_analyze_sample_destrule() {
+istioctl analyze samples/bookinfo/networking/bookinfo-gateway.yaml samples/bookinfo/networking/destination-rule-all.yaml
+}
+
+snip_analyze_networking_directory() {
+istioctl analyze samples/bookinfo/networking/
+}
+
+snip_analyze_all_networking_yaml() {
+istioctl analyze samples/bookinfo/networking/*.yaml
+}
+
+snip_analyze_all_networking_yaml_no_kube() {
+istioctl analyze --use-kube=false samples/bookinfo/networking/*.yaml
+}
+
+! read -r -d '' snip_vs_yaml_with_status <<\ENDSNIP
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  annotations:
+    kubectl.kubernetes.io/last-applied-configuration: |
+      {"apiVersion":"networking.istio.io/v1alpha3","kind":"VirtualService","metadata":{"annotations":{},"name":"ratings","namespace":"default"},"spec":{"hosts":["ratings"],"http":[{"route":[{"destination":{"host":"ratings","subset":"v1"}}]}]}}
+  creationTimestamp: "2019-09-04T17:31:46Z"
+  generation: 11
+  name: ratings
+  namespace: default
+  resourceVersion: "12760039"
+  selfLink: /apis/networking.istio.io/v1alpha3/namespaces/default/virtualservices/ratings
+  uid: dec86702-cf39-11e9-b803-42010a8a014a
+spec:
+  gateways:
+  - bogus-gateway
+  hosts:
+  - ratings
+  http:
+  - route:
+    - destination:
+        host: ratings
+        subset: v1
+status:
+  validationMessages:
+  - code: IST0101
+    level: Error
+    message: 'Referenced gateway not found: "bogus-gateway"'
```

Comment:
A common approach to make this testable, and also more readable in the doc, is to change the "noise" to "...". You need to change it in the .md file and then regen the snip which will then look something like this:

```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
...
spec:
  gateways:
  - bogus-gateway
  hosts:
  - ratings
  http:
  - route:
    - destination:
        host: ratings
        subset: v1
status:
  validationMessages:
  - code: IST0101
    level: Error
    message: 'Referenced gateway not found: "bogus-gateway"'
```

Then, in the test you can do something like this:

```
get_ratings_virtual_service() {
kubectl get vs ratings -o yaml
}
_verify_elided get_ratings_virtual_service "$snip_vs_yaml_with_status"
```